### PR TITLE
Remove unused function bn_print_fp()

### PR DESCRIPTION
**Pull request by ni4**
Just some minor cleanup.

### CODE REVIEW

File: src/lib/crypto/bn.cpp [hunk 1]
```diff
@@ -39,38 +37,27 @@
 bignum_t *
 bn_bin2bn(const uint8_t *data, int len, bignum_t *ret)
 {
-    if (data == NULL) {
+    if (!data || !ret) {
```
Comment (dewyatt) [inline src/lib/crypto/bn.cpp:40]: Just a heads up, I don't think this is right, I think the original intention here was:
* if `!data` return NULL, as that's clearly a mistake. (should be an assertion here too)
* if `!ret` allocate a `bn` and use that for the function and return it
* if `ret` then use that `bn` and return it

Comment (ni4) [reply src/lib/crypto/bn.cpp:40]: @dewyatt Thanks, mislooked this. Re-pushing. Will RNP_LOG() be enough as we don't use assert's often?

Comment (ni4) [reply src/lib/crypto/bn.cpp]: Ok, let's add an assertion as well.

File: src/lib/crypto/bn.cpp [hunk 2]
```diff
@@ -27,10 +27,9 @@
 #include "bn.h"
 #include <botan/ffi.h>
 #include <stdlib.h>
+#include <assert.h>
```
Comment (dewyatt) [inline src/lib/crypto/bn.cpp:30]: I think technically these C headers are deprecated and we should be using `cassert`, but I know we're doing this all over the codebase and it's maybe not a concern right now

**Review by dewyatt — APPROVED** [review]

File: src/lib/crypto/bn.cpp (hunk 2 above)
Comment (ni4) [reply src/lib/crypto/bn.cpp:30]: You're right, just sometimes hard to think C++ when surrounded by the C code :)

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thanks guys!

**Pull request merged**